Extension/src/LanguageServer/client.ts

```diff
@@ -2348,6 +2348,27 @@ export class DefaultClient implements Client {
         return result;
     }
 
+    // This is used to avoid processing unnecessary LSP cancel requests during a Copilot completion
+    // when the result could still be used in 2 minutes in its cache before it sends another request.
+    private static async withoutLspCancellationHandling<T>(task: () => Promise<T>): Promise<T> {
+        let result: T;
+
+        try {
+            result = await task();
+        } catch (e: any) {
+            if (e instanceof ResponseError && e.code === ServerCancelled) {
+                // This can occur if the user switches files and makes edits fast enough.
+                // It doesn't seem like a very important scenario, so cancelling seems fine, even though
+                // not cancelling would allow the value to be cached if the user switches back to that file.
```

Comment:
FYI, the entries in the cache are per URI. The sooner an entry is computed, the better. Hence, if we have an actual result, we should always return it back regardless of the cancellation.

Reply:
@lukka This was the cancellation code in cpptools that I had removed, because you said you wanted it to stay in case there were multiple requests sent. I can certainly revive that PR.

Reply:
@sean-mcmanus No need to revive that PR, I am just sharing what I had in my mind when I was readind that comment, which looks like saying that the cache is a single entry, although I may have misunderstood.

Reply:
Well, assuming that's what you want, I'll cancel this PR and reopen the abandoned one.